classes/requests/helpers/class-dintero-checkout-cart.php

```diff
@@ -0,0 +1,287 @@
+		}
+
+		return $order_lines;
+	}
+
+	/**
+	 * Retrieve all the cart items (excluding shipping).
+	 *
+	 * @return void Populates $this->products.
+	 */
+	private function order_items() {
+		$cart = WC()->cart->get_cart();
+
+		foreach ( $cart as $item ) {
+			$id      = ( empty( $item['variation_id'] ) ) ? $item['product_id'] : $item['variation_id'];
+			$product = wc_get_product( $id );
+
+			$order_item = array(
+				/* NOTE: The id and line_id must match the same id and line_id on capture and refund. */
+				'id'          => strval( $id ),
+				'line_id'     => strval( $id ),
+				'description' => $product->get_name(),
+				'quantity'    => $item['quantity'],
+				'amount'      => intval( number_format( $item['line_total'] * 100, 0, '', '' ) ),
+				'vat_amount'  => intval( number_format( $item['line_tax'] * 100, 0, '', '' ) ),
+			);
+
+			if ( $product->is_taxable() ) {
+				$tax_rate          = WC_TAX::get_base_tax_rates( $product->get_tax_class() );
+				$order_item['vat'] = empty( $tax_rate ) ? 0 : reset( $tax_rate )['rate'];
+			}
+
+			$order_item['amount'] += $order_item['vat_amount'];
+			$this->products[]      = $order_item;
+		}
+
+	}
+
+	/**
+	 * Retrieve all the discount items.
+	 *
+	 * @return void Populates $this->discounts.
+	 */
+	private function discount_items() {
+
+		foreach ( WC()->cart->get_coupons() as $coupon_code => $coupon ) {
+			$discount_item            = array(
+				'amount'        => intval( number_format( $coupon->get_amount() * 100, 0, '', '' ) ),
+				'discount_id'   => strval( $coupon_code ),
+				'discount_type' => 'manual',
+				'description'   => $coupon->get_description(),
+				/* line_id must be an integer. */
+				'line_id'       => $coupon->get_id(),
+			);
+			$discount_item['amount'] += intval( number_format( WC()->cart->get_coupon_discount_tax_amount( $coupon_code ) * 100, 0, '', '' ) );
+
+			array_push( $this->discounts['discount_lines'], $discount_item );
+			array_push( $this->discounts['discount_codes'], $coupon_code );
+
+		}
+	}
+
+	/**
+	 * Retrieve all the fee items.
+	 *
+	 * @param WC_Order $order WooCommerce Order.
+	 * @return void Populates $this->fees.
+	 */
+	private function fee_items( $order ) {
+
+		foreach ( $order->get_fees() as $fee ) {
+			$name     = $fee->get_name();
+			$fee_item = array(
+				/* NOTE: The id and line_id must match the same id and line_id on capture and refund. */
+				'id'          => $name,
+				'line_id'     => $name,
+				'description' => $name,
+				'quantity'    => $fee->get_quantity(),
+				'amount'      => intval( number_format( $fee->get_total() * 100, 0, '', '' ) ),
+				'vat_amount'  => intval( number_format( $fee->get_total_tax() * 100, 0, '', '' ) ),
+			);
+
+			$fee_item['amount'] += $fee_item['vat_amount'];
+			$fee_item['vat']     = ( ! empty( $fee_item['vat_amount'] ) ) ? intval( number_format( $fee_item['vat_amount'] / $fee_item['amount'] * 100, 0, '', '' ) ) : 0;
```

Comment:
I think it would be smart to extract all money-logic into it's own class

Reply:
This way you could also change from 100 to another currency's preferred number of decimals without changing it all over.

Reply:
This way you could also change from 100 to another currency's preferred number of decimals without changing it all over.

Reply:
For example, this complex logic is repeated in `shipping_options`. Would be nice to not have to change it multiple places.

Reply:
I definitely agree, and I did consider it, but I did not want to stray too far away from how my colleagues have written the cart class in our other plugins (i.e., the coding style) to make troubleshooting/code review easier.

Since you also noticed this, I'll bring this up to them.

Reply:
utility functions file sounds good :)

Reply:
Great! I'll make sure to add this :+1: